src/vs/workbench/browser/parts/activitybar/activitybarActions.ts

```diff
@@ -131,36 +137,56 @@ export class GlobalActivityActionItem extends ActivityActionItem {
 		this._register(DOM.addDisposableListener(this.container, DOM.EventType.MOUSE_DOWN, (e: MouseEvent) => {
 			DOM.EventHelper.stop(e, true);
 
-			const event = new StandardMouseEvent(e);
-			this.showContextMenu({ x: event.posx, y: event.posy });
+			if (this.menuVisible) {
+				this.hideContextMenu();
+			} else {
+				this.showContextMenu();
+			}
 		}));
 
 		this._register(DOM.addDisposableListener(this.container, DOM.EventType.KEY_UP, (e: KeyboardEvent) => {
 			let event = new StandardKeyboardEvent(e);
 			if (event.equals(KeyCode.Enter) || event.equals(KeyCode.Space)) {
 				DOM.EventHelper.stop(e, true);
 
-				this.showContextMenu(this.container);
+				if (this.menuVisible) {
+					this.hideContextMenu();
+				} else {
+					this.showContextMenu();
+				}
 			}
 		}));
 
 		this._register(DOM.addDisposableListener(this.container, TouchEventType.Tap, (e: GestureEvent) => {
 			DOM.EventHelper.stop(e, true);
 
-			const event = new StandardMouseEvent(e);
-			this.showContextMenu({ x: event.posx, y: event.posy });
+			if (this.menuVisible) {
+				this.hideContextMenu();
+			} else {
+				this.showContextMenu();
+			}
 		}));
 	}
 
-	private showContextMenu(location: HTMLElement | { x: number, y: number }): void {
+	private hideContextMenu(): void {
+		if (this.menuVisible) {
+			this.menuVisible = false;
+			this.contextMenuService.hideContextMenu();
+		}
+	}
+
+	private showContextMenu(): void {
 		const globalAction = this._action as GlobalActivityAction;
 		const activity = globalAction.activity as IGlobalActivity;
 		const actions = activity.getActions();
+		const containerPosition = DOM.getDomNodePagePosition(this.container);
```

Comment:
Is this a fix that should be made separately, it seems to be around positioning the menu?

Reply:
it can be separated if needed. I was in the code area and the same menu had 2 issues